neuro_san/session/async_direct_agent_session.py

```diff
@@ -170,6 +175,10 @@ async def streaming_chat(self, request_dict: Dict[str, Any]) -> Generator[Dict[s
             response_dict: Dict[str, Any] = copy(template_response_dict)
             if message_filter.allow(message):
                 # We expect the message to be a dictionary form of chat.ChatMessage
+                if message_processor is not None:
+                    message_type: ChatMessageType = message.get("type")
+                    # Can modify message
+                    message_processor.process_message(message, message_type)
```

Comment:
If the front man says we want to parse out structure formats do that for every message.